src/SecureCommunication/Secio1.cs

```diff
@@ -56,21 +53,21 @@ public async Task<Stream> EncryptAsync(PeerConnection connection, CancellationTo
             rng.NextBytes(localNonce);
             var localProposal = new Secio1Propose
             {
-                Nonce = localNonce,
+                Nonce = ByteString.CopyFrom(localNonce),
                 Exchanges = "P-256,P-384,P-521",
                 Ciphers = "AES-256,AES-128",
                 Hashes = "SHA256,SHA512",
-                PublicKey = Convert.FromBase64String(localPeer.PublicKey)
+                PublicKey = ByteString.FromBase64(localPeer.PublicKey),
             };
 
-            ProtoBuf.Serializer.SerializeWithLengthPrefix(stream, localProposal, PrefixStyle.Fixed32BigEndian);
+            localProposal.WriteDelimitedTo(stream);
```

Comment:
`WriteDelimitedTo` writes the length as varint32. The protocol requires `Fixed32BigEndian`.

Reply:
Got it